.github/workflows/ci.yml

```diff
@@ -149,16 +162,25 @@ jobs:
       VS_GENERATOR_TOOLSET: ${{ matrix.VS_GENERATOR_TOOLSET }}
 
     steps:
+      - name: Installing Alpine Linux Dependencies
+        if: ${{ contains(matrix.container, 'alpine') }}
+        run: |
+          apk update
+          apk add bash build-base cargo cmake curl curl-dev git icu linux-headers mitmproxy perl python3-dev sudo ${{ matrix.APK_PACKAGES }}
+          curl -sSL --retry 5 https://dot.net/v1/dotnet-install.sh | bash -eo pipefail /dev/stdin --channel 8.0 --install-dir /usr/share/dotnet
+          ln -s /usr/share/dotnet/dotnet /usr/local/bin/dotnet
+
       - uses: actions/checkout@v4
         with:
           submodules: recursive
       - uses: actions/setup-python@v5
+        if: ${{ !matrix.container }}
```

Comment:
`setup-python` doesn't work in arbitrary containers but basically only supports the runner images: https://github.com/actions/python-versions

Reply:
Ugh, the container integration of GHA workflows is a little weird, to be honest. 

Two things: 

`setup-python` is only necessary because we haven't switched to using `uv` for the test setup yet. So this is not a long-term special case. 

Suppose the container setup (essentially, now parallel conditional build-steps in the workflow yaml) gets out of hand at some point. In that case, we can also evaluate switching to Dockerfiles written on top of the Alpine image, manually creating images that already contain what we want, and executing the pytest there.